### PR TITLE
Image similarity vision feature print screen

### PR DESCRIPTION
**Pull request by TobyRoseman**
Fixes unreliable Image Similarity unit test by using a better comparison metric.
Adds vision feature print screen to image similarity (#705)

### CODE REVIEW

File: src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py [hunk 1]
```diff
@@ -48,8 +49,13 @@ def create(dataset, label = None, feature = None, model = 'resnet-50', verbose =
         Uses a pretrained model to bootstrap an image similarity model
 
            - "resnet-50" : Uses a pretrained resnet model.
+
            - "squeezenet_v1.1" : Uses a pretrained squeezenet model.
 
+           - "VisionFeaturePrint_Screen": Uses an OS internal feature extractor.
+                                          Only on available on iOS,tvOS 12.0+,
```
[the remark this answers is not in the document]
Comment (TobyRoseman) [reply src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py]: I believe the tvOS version is also 12.0+

Would the following be more clear? `Only on available on iOS 12.0+, macOS 10.14+, and tvOS 12.0+.`

[the remark this answers is not in the document]
Comment (znation) [reply src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py]: Oh, you're right, it is. Sorry I assumed the earlier one was a typo. I do think it is more clear the way you're suggesting.

File: src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py [hunk 2]
```diff
@@ -201,6 +214,9 @@ def _load_version(cls, state, version):
         _tkutl._model_version_check(version, cls._PYTHON_IMAGE_SIMILARITY_VERSION)
         from turicreate.toolkits.nearest_neighbors import NearestNeighborsModel
         state['similarity_model'] = NearestNeighborsModel(state['similarity_model'])
+        if state['model'] == "VisionFeaturePrint_Screen" and _mac_ver() < (10,14):
+            raise ToolkitError("Can not load model on this operating system. This model uses VisionFeaturePrint_Screen, " \
```
Comment (gustavla) [inline src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py]: Nit: Note that `\` is not necessary inside parens.

File: src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py [hunk 3]
```diff
@@ -491,40 +507,84 @@ def export_coreml(self, filename):
...
+            DOUBLE_ARRAY_VALUE = _cmt.proto.FeatureTypes_pb2.ArrayFeatureType.ArrayDataType.Value('DOUBLE')
+
+            top_spec = _cmt.proto.Model_pb2.Model()
+            top_spec.specificationVersion = 3
+            desc = top_spec.description
+
+            input = desc.input.add()
+            input.name = self.feature
+            input.type.imageType.width = 299
+            input.type.imageType.height = 299
```
Comment (gustavla) [inline src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py]: Do we need to repeat 299 here? We already stored this inside `input_image_shape`.

File: src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py (hunk 2 above)
Comment (gustavla) [reply src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py]: No need to change, just an FYI.

File: src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py [hunk 4]
```diff
@@ -491,40 +507,84 @@ def export_coreml(self, filename):
...
+            pipeline = top_spec.pipeline
+            scene_print = pipeline.models.add()
+            scene_print.specificationVersion = 3
+            scene_print.visionFeaturePrint.scene.version = 1
+
+            input = scene_print.description.input.add()
+            input.name = self.feature
+            input.type.imageType.width = 299
+            input.type.imageType.height = 299
+            input.type.imageType.colorSpace = BGR_VALUE
```
Comment (gustavla) [inline src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py]: Do all image feature extractors in TC use BGR? I'm looking inside `mlmodel_image_feature_extractor.mm` and that seems to be the case. So are all the MXNet models also trained on BGR?

File: src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py (hunk 2 above)
Comment (TobyRoseman) [reply src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py]: Interesting. I didn't know that. I updated here and in the image classifier.

File: src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (TobyRoseman) [reply src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py]: No worries. The wording is confusing. I've updated it here and in the image classifier.

File: src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py (hunk 3 above)
Comment (TobyRoseman) [reply src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py]: Ok. Done.

File: src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py (hunk 4 above)
Comment (TobyRoseman) [reply src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py]: I'm not sure. I believe the all the Core ML models used by image classification/similarity use BGR. I believe the MxNet models use RGB.

**Pull request merged**